### PR TITLE
Extra examples

### PR DESCRIPTION
**Pull request by pdellaert**
Two complex example scripts using pyvmomi to connect to vCenter and one
simple example for a full overview of all enterprises with all domains,
zones, subnets and acls

### CODE REVIEW

File: deploy_vsphere_template_with_nuage.py [hunk 1]
```diff
@@ -0,0 +1,648 @@
+"""
```
Comment (primalmotion) [inline deploy_vsphere_template_with_nuage.py]: adding `# -*- coding: utf-8 -*-` as first line is a good practice

File: deploy_vsphere_template_with_nuage.py [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+    parser.add_argument('--vcenter-host', nargs=1, required=True, help='The vCenter or ESXi host to connect to', dest='vcenter_host', type=str)
+    parser.add_argument('--vcenter-port', nargs=1, required=False, help='vCenter Server port to connect to (default = 443)', dest='vcenter_port', type=int, default=[443])
+    parser.add_argument('--vcenter-password', nargs=1, required=False, help='The password with which to connect to the vCenter host. If not specified, the user is prompted at runtime for a password', dest='vcenter_password', type=str)
+    parser.add_argument('--vcenter-user', nargs=1, required=True, help='The username with which to connect to the vCenter host', dest='vcenter_username', type=str)
+    parser.add_argument('-v', '--verbose', required=False, help='Enable verbose output', dest='verbose', action='store_true')
+    
+    args = parser.parse_args()
+    return args
+
+def cls(logger):
```
Comment (primalmotion) [inline deploy_vsphere_template_with_nuage.py]: cls is not strictly enforced as a reserved word, but it should be. Can you renamed that as `clear()`

File: deploy_vsphere_template_with_nuage.py [hunk 3]
```diff
@@ -0,0 +1,648 @@
...
+    """
+    Clears the terminal
+    """
+    if logger:
+        logger.debug('Clearing terminal')
+    os.system(['clear','cls'][os.name == 'nt'])
+
+def calc_cidrmask(netmask,logger):
```
Comment (primalmotion) [inline deploy_vsphere_template_with_nuage.py]: there is a very cool library called `ipaddress` which should do whatever you need regarding the ip. I believe its a standard module in Python 3 but can be pip installed.

What about using it?

File: deploy_vsphere_template_with_nuage.py [hunk 4]
```diff
@@ -0,0 +1,648 @@
...
+   """
+   if logger:
+        logger.debug('Checking if IP %s is inside subnet %s' % (ip,net_n_bits))
+   ipaddr = struct.unpack('<L', socket.inet_aton(ip))[0]
+   net, bits = net_n_bits.split('/')
+   netaddr = struct.unpack('<L', socket.inet_aton(net))[0]
+   netmask = ((1L << int(bits)) - 1)
+   return ipaddr & netmask == netaddr & netmask
+
+def find_vm(vc,logger,name):
```
Comment (primalmotion) [inline deploy_vsphere_template_with_nuage.py]: I know I'm picky, but there should be a space after a `,`

Like 

```python
def find_vm(vc, logger, name):
```

Comment (pdellaert) [inline deploy_vsphere_template_with_nuage.py]: Picky is fine :). Let me look at it tomorrow ;). 

With the ipaddress package, i didn't use it, as it's indeed in 3, not sure about pip, but i can have a look at it as well.

File: deploy_vsphere_template_with_nuage.py [hunk 5]
```diff
@@ -0,0 +1,648 @@
...
+    """
+
+    for cur_ent in nc.user.enterprises.get():
+        logger.debug('Checking Nuage enterprise %s' % cur_ent.name)
+        if cur_ent.name == name:
+            logger.debug('Found Nuage enterprise %s' % name)
+            return cur_ent
+    return None
+
+def find_user(enterprise,logger,name):
```
Comment (primalmotion) [inline deploy_vsphere_template_with_nuage.py]: why don't you use the filter for all of these functions?

```python
def find_user(enterprise, logger, username):
    return enterprise.users.get_first(filter="userName == '%s' % username)
```

That way you don't even need all that helpers

**Comment by pdellaert** [comment]
All changes were implemented, and i updated to the latest VSPK version (build with Monolithe/vspkgenerator)

**Comment by pdellaert** [comment]
Any status on this request?

**Comment by primalmotion** [comment]
All good :)

Thanks

**Pull request merged**